--- decompiled/TerrariaUnified/Terraria/Projectile.cs
+++ decompiled/Wardian/Terraria/Projectile.cs
@@ -29325,7 +_,7 @@
 				}
 			}
 			float amount = MathHelper.Lerp(value, value2, Utils.GetLerpValue(num, 30f, timeLeft, clamped: true));
-			velocity = Vector2.SmoothStep(velocity, value3, amount);
+			velocity = NumericsExtensions.SmoothStep(velocity, value3, amount);
 			if (type == 931) {
 				velocity *= MathHelper.Lerp(0.85f, 1f, Utils.GetLerpValue(0f, 90f, timeLeft, clamped: true));
 			}
@@ -30118,7 +_,7 @@
 		Vector2 vector2 = -velocity;
 		Vector2 value = center + vector2 * 2f;
 		Vector2 value2 = vector + vector2 * (1f - num2 * 3f);
-		base.Center = Vector2.CatmullRom(value, center, vector, value2, num2);
+		base.Center = NumericsExtensions.CatmullRom(value, center, vector, value2, num2);
 		int num3 = type;
 		if (num3 == 856) {
 			Lighting.AddLight(base.Center, Color.HotPink.ToVector3() * 0.3f);
@@ -30249,7 +_,7 @@
 			ai[1] = 60f;
 		}
 		ai[0] += 1f / ai[1];
-		Vector2 vector3 = Vector2.CatmullRom(amount: Utils.Clamp(ai[0], 0f, 1f), value1: value, value2: vector, value3: vector2, value4: value2);
+		Vector2 vector3 = NumericsExtensions.CatmullRom(amount: Utils.Clamp(ai[0], 0f, 1f), value1: value, value2: vector, value3: vector2, value4: value2);
 		if (flag) {
 			for (float num3 = 0f; num3 < 1f; num3 += 0.5f) {
 				if (Main.rand.Next(19) == 0) {
@@ -32813,7 +_,7 @@
 				AI_GetMyGroupIndexAndFillBlackList(blacklist, out var index2, out var totalIndexesInGroup2);
 				AI_156_GetIdlePosition(index2, totalIndexesInGroup2, out var idleSpot2, out var _);
 				velocity = Vector2.Zero;
-				base.Center = Vector2.SmoothStep(base.Center, idleSpot2, 0.45f);
+				base.Center = NumericsExtensions.SmoothStep(base.Center, idleSpot2, 0.45f);
 				if (Main.rand.Next(20) == 0) {
 					int num6 = AI_156_TryAttackingNPCs(blacklist);
 					if (num6 != -1) {
@@ -32831,7 +_,7 @@
 			AI_GetMyGroupIndexAndFillBlackList(blacklist, out var index3, out var totalIndexesInGroup3);
 			AI_156_GetIdlePosition(index3, totalIndexesInGroup3, out var idleSpot3, out var idleRotation3);
 			velocity = Vector2.Zero;
-			base.Center = Vector2.SmoothStep(base.Center, idleSpot3, 0.45f);
+			base.Center = NumericsExtensions.SmoothStep(base.Center, idleSpot3, 0.45f);
 			rotation = rotation.AngleLerp(idleRotation3, 0.45f);
 			if (Main.rand.Next(20) == 0) {
 				int num7 = AI_156_TryAttackingNPCs(blacklist);
@@ -50916,7 +_,7 @@
 							for (int num831 = 0; num831 < 2; num831++) {
 								Vector2 value12 = vector69.RotatedBy(((num831 == 0) ? 1f : (-1f)) * ((float)Math.PI * 2f) / (num826 * 2f));
 								for (float num832 = 0f; num832 < num829; num832++) {
-									Vector2 vector70 = Vector2.SmoothStep(vector69, value12, num832 / num829);
+									Vector2 vector70 = NumericsExtensions.SmoothStep(vector69, value12, num832 / num829);
 									float num833 = MathHelper.SmoothStep(num827, num828, num832 / num829);
 									int num834 = Dust.NewDust(new Vector2(position.X, position.Y), 6, 6, 134, 0f, 0f, 100, default(Color), 1.3f);
 									Dust dust2 = Main.dust[num834];
